### PR TITLE
PSP-9452: Split map search for PID PIN

### PR DESCRIPTION
**Pull request by asanchezr**
map-search

listview-search

### CODE REVIEW

File: source/backend/dal/Helpers/Extensions/PropertyViewExtensions.cs [hunk 1]
```diff
@@ -54,22 +51,27 @@ public static class PropertyViewExtensions
...
+            {
+                // note - 2 part search required. all matches found by removing leading 0's, then matches filtered in subsequent step. This is because EF core does not support an lpad method.
+                Regex nonInteger = new Regex("[^\\d]");
+                var formattedPid = Convert.ToInt32(nonInteger.Replace(filter.Pid, string.Empty)).ToString();
+                predicateBuilder = predicateBuilder.And(p => EF.Functions.Like(p.Pid.ToString(), $"%{formattedPid}%"));
+            }
+
+            if (!string.IsNullOrWhiteSpace(filter.Pin))
             {
                 // note - 2 part search required. all matches found by removing leading 0's, then matches filtered in subsequent step. This is because EF core does not support an lpad method.
```
Comment (devinleighsmith) [inline source/backend/dal/Helpers/Extensions/PropertyViewExtensions.cs:71]: this comment is no longer applicable here, correct?

File: source/backend/dal/Repositories/PropertyRepository.cs [hunk 2]
```diff
@@ -56,25 +56,32 @@ public int Count()
...
 
-            if (!string.IsNullOrWhiteSpace(filter.PinOrPid))
+            if (!string.IsNullOrWhiteSpace(filter.Pid))
+            {
+                Regex nonInteger = new Regex("[^\\d]");
+                var formattedPid = nonInteger.Replace(filter.Pid, string.Empty);
+                items = items.Where(i => i.Pid.ToString().PadLeft(9, '0').Contains(formattedPid)).ToArray();
+            }
+
+            if (!string.IsNullOrWhiteSpace(filter.Pin))
```
Comment (devinleighsmith) [inline source/backend/dal/Repositories/PropertyRepository.cs:80]: is this necessary? it seems like prior to this there was some lpadding happening for the pid/pin, but it seems like a 2 part search isn't required for pin.

File: source/backend/tests/core/PimsTestContext.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+using Microsoft.Extensions.Options;
+using Pims.Dal;
+using Pims.Dal.Entities;
+
+namespace Pims.Core.Test
+{
+    /// <summary>
+    /// PimsTestContext class, provides a data context to manage the datasource for the PIMS application (specific to UNIT TESTS).
+    /// </summary>
+    public class PimsTestContext : PimsContext
```
Comment (devinleighsmith) [inline source/backend/tests/core/PimsTestContext.cs:13]: thanks for this.

**Review by devinleighsmith — APPROVED** [review]

File: source/backend/dal/Helpers/Extensions/PropertyViewExtensions.cs (hunk 1 above)
Comment (asanchezr) [reply source/backend/dal/Helpers/Extensions/PropertyViewExtensions.cs:71]: Not sure if it applies to be honest. I think you implemented the original logic and I didn't want to break it.
If you say that the padding applies to PID only then I will remove the 2-part search from the PIN option (leaving PID as-is)

File: source/backend/dal/Repositories/PropertyRepository.cs (hunk 2 above)
Comment (asanchezr) [reply source/backend/dal/Repositories/PropertyRepository.cs:80]: I see your point - I will remove the 2-part search for PIN (and leave PID as-is)

**Pull request merged**